tests/unit/source_code/notebooks/test_sources.py

```diff
@@ -113,3 +116,241 @@ def test_file_linter_lints_file_with_missing_read_permission(migration_index, mo
     assert len(advices) == 1
     assert advices[0].code == "file-permission"
     assert advices[0].message == f"Missing read permission for {path}"
+
+
+class _NotebookLinter(NotebookLinter):
+    """A helper class to construct the notebook linter from source code for testing simplification."""
+
+    @classmethod
+    def from_source_code(
+        cls,
+        index: TableMigrationIndex,
+        path_lookup: PathLookup,
+        session_state: CurrentSessionState,
+        source: str,
+        default_language: Language,
+    ) -> NotebookLinter:
+        ctx = LinterContext(index)
+        notebook = Notebook.parse(Path(""), source, default_language)
+        assert notebook is not None
+        return cls(ctx, path_lookup, session_state, notebook)
+
+
+def test_notebook_linter_lints_source_yielding_no_advices(migration_index, mock_path_lookup) -> None:
+    linter = _NotebookLinter.from_source_code(
+        migration_index,
+        mock_path_lookup,
+        CurrentSessionState(),
+        "# Databricks notebook source\nprint(1)",
```

Comment:
Would like there to be a trailing newline there…
```suggestion
        "# Databricks notebook source\nprint(1)\n",
```